HW #1 (Linter and Code Review)

cholombytko opened the pull request:
Changes made:
- Updated ESLint config
- Implemented Github Actions workflow
- Fixed errors which occured during linting

cholombytko committed:
feat: add new ESLint rules for code styling

Changes include:
- Installed "perfectionist" package.
- Updated .eslintrc using perfectionist rules.

cholombytko committed:
feat: NodeJS lint implementation

This creates automatic linting workflow using Github Actions by creating
ci.yml

cholombytko committed:
fix: delete automated fixing during linting

This fixes automated fixing which helps to define errors during lint
workflow

cholombytko committed:
fix: fix errors defined by linter

This fixes every error which was defined by linter during lint workflow

eluyema commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,24 @@
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
```
You can add caching for dependencies, it's make your workflow faster for a 5-7 seconds)
Just add new line `cache: 'npm'` after line `node-version: 20`
Here readme of actions/setup-node with example: https://github.com/actions/setup-node#caching-global-packages-data

cholombytko replied:
This is a good idea, got it

eluyema commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,24 @@
+          node-version: 20
+
+      - name: Install ESLint
+        run: npm install eslint
```
Using` npm install` can invoke unexpected behavior because it might install newer versions of ESLint or some ESLint plugins that differ from those you use locally. It's better to replace it with `npm ci`

eluyema commented on package.json:
```diff
@@ -14,7 +14,7 @@
     "migration:run": "npm run typeorm -- migration:run",
     "start:prod": "npm run migration:run && node dist/main",
-    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
+    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\"",
```
You can add this params in the and of eslint command - `--ignore-path .gitignore`. For now, it's not required, but it's good practice to add this params for eslint command.

eluyema commented on src/database.module.ts:
```diff
@@ -8,19 +8,19 @@ import { TypeOrmModule } from '@nestjs/typeorm';
       imports: [ConfigModule],
```
Maybe it would be good to have folder for each module? Your modules app.module, database.module, mail.module don't have own folder, but another module have own folder. It's just folder structure issues, so if it's your style, than you can stay it as is.

eluyema commented on src/subscribe/subscribe.service.ts:
```diff
@@ -1,9 +1,9 @@
+import { MailingService } from 'src/email/services/mailing.service';
 
 @Injectable()
 export class SubscribeService {
```
In line 19, one letter s in the word successfully is missing

eluyema commented on src/subscribe/subscribe.service.ts:
```diff
@@ -1,9 +1,9 @@
-import { Injectable } from '@nestjs/common';
-import { EmailService } from 'src/email/services/email.service';
-import { MailingService } from 'src/email/services/mailing.service';
 import { ICreateSubscription } from './interfaces/create-subscription.interface';
```
I think it would be better to rename **subscribe** (verb) module and service to **subscription** module and service. verb is not very convenient looks for this names, because for another modules already be used noun for names.

eluyema commented:
In Dockerfile, it seems that you need to use npm run start:prod, instead of start:dev

eluyema commented:
That's all the issues I found, looks good :)

eluyema commented:
If you have any questions to my comments, you can add it to review's comment and I will answer you

cholombytko committed:
ci: add npm ci and npm caching

This commit updates the CI configuration.
Changes made:
- add npm ci for clean install
- add cache: 'npm' for improving ci workflow speed

cholombytko committed:
style: fix typo

Larisa-Antsifrova commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,25 @@
+        run: npm ci
+
+      - name: Run ESLint
+        run: npm run lint
```
It's nice to see that you've completed the extra task as well! Thank you 💪

cholombytko committed:
fix: change subscribe verb to noun subscription
